### PR TITLE
Cache completion based on XML Schema/DTD

### LINKED ISSUE #547: Cache completion based on XML Schema/DTD
When completion based on XML Schema/DTD is done and XML Schema/DTD  comes from the file system:

 * the DTD/Schema URI use file system
 * the DTD/Schema URI use http but cache is activated

The XMl Schema / DTD is every time parsed and loaded. For a big XML Schema / DTD, there are some trouble:

 * completion can be slow.
 * if it's too slow, completion returns nothing.

To fix this issue, we should cache the completion (the CMDocument) like we have done for http but we should track all files changed of the XML Schema (which can imported other XML Schema) to clean the cache when the XMl Schema / DTD is modified.

### PR DESCRIPTION
**Pull request by angelozerr**
Fixes #547

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+	}
+
+	private final List<FileChangedTracker> files;
+
+	public FilesChangedTracker() {
+		files = new ArrayList<>();
+	}
+
+	public void addFileURI(String fileURI) {
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: you need a way to remove tracked files

Comment (angelozerr) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: why?

**Comment by angelozerr** [comment]
@fbricon working on test

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java (hunk 1 above)
Comment (fbricon) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: to prevent memory leaks?

Comment (angelozerr) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: The track is very simply, it checks just the last modified of file eeach time. I don't use watcher files system.

Comment (fbricon) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: so what happens if the tracked file is deleted?

Comment (angelozerr) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: Good catch! XSD Schema was stored in memory and completion continued to work. It should not. I have fixed the problem and cover that in my test.

**Comment by angelozerr** [comment]
@fbricon all tests was done, is there something that you wish I do to merge this PR?

**Pull request merged**